storage.bs

```diff
@@ -94,37 +100,251 @@ function retrieveNextChunk(nextChunkInfo) {
+infrastructure defined by this standard to keep track of its storage needs.
+
+<p>A <a>storage endpoint</a> has an <dfn for="storage endpoint">identifier</dfn>, which is a
+<a>storage identifier</a>.
+
+<p>A <a>storage endpoint</a> also has <dfn for="storage endpoint">types</dfn>, which is a
+<a for=/>set</a> of <a>storage types</a>.
+
+<p>A <dfn>storage identifier</dfn> is an <a for=/>ASCII string</a>.
+
+<hr>
+
+<p>The <dfn>registered storage endpoints</dfn> are a <a for=/>set</a> of <a>storage endpoints</a>
+defined by the following table:
+
+<table>
+ <tr>
+  <th><a>Storage identifier</a>
+  <th><a>Storage types</a>
+ <tr>
+  <td>"<code>caches</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>indexedDB</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>localStorage</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>serviceWorkerRegistrations</code>"
+  <td>« "<code>storage</code>" »
+ <tr>
+  <td>"<code>sessionStorage</code>"
+  <td>« "<code>session-storage</code>" »
+</table>
+
+
+
+<h3 id=buckets oldids=boxes>Storage buckets</h3>
+
+<p>An <dfn>abstract storage bucket</dfn> is a place for <a>storage endpoints</a> to store data.
+Whenever an <a>abstract storage bucket</a> is cleared by the user agent, it must be cleared in its
+entirety.
+<!-- This is probably better defined in some kind of storage lifetime section. -->
+
+<p>An <a>abstract storage bucket</a> has a
+<dfn for="abstract storage bucket,storage bucket,session storage bucket">map</dfn> of
+<a>storage identifiers</a> to <a for="abstract storage bucket">areas</a>.
 
-<p><a>Storage</a> consists of zero or more <dfn export id=site-storage-unit>storage units</dfn>.
+<p>An <dfn for="abstract storage bucket,storage bucket,session storage bucket">area</dfn> is a part
+of an <a>abstract storage bucket</a> carved out for a <a>storage endpoint</a>. An
+<a for="abstract storage bucket">area</a> has a
+<dfn for="abstract storage bucket/area,storage bucket/area,session storage bucket/area">map</dfn>,
+which is initially an empty <a for=/>map</a>. An <a for="abstract storage bucket">area</a> also has
+a
+<dfn for="abstract storage bucket/area,storage bucket/area,session storage bucket/area">proxy map reference set</dfn>,
+which is initially an empty <a for=/>set</a>.
 
-<p>Each <a for=/>origin</a> has an associated <a>storage unit</a>. A <a>storage unit</a> contains a
-single <dfn export id=bucket oldids=box>bucket</dfn>. [[HTML]]
+<p>How an <a>abstract storage bucket</a>'s <a for="abstract storage bucket">area</a>'s
+<a for="abstract storage bucket/area">map</a> is stored is <a>implementation-defined</a>. How it is
+made available across <a>agent</a> or even <a>agent cluster</a> boundaries is
+<a>implementation-defined</a>.
 
+<hr>
 
-<h3 id=buckets oldids=boxes>Buckets</h3>
+<p>A <dfn id=bucket oldids=box>storage bucket</dfn> is an <a>abstract storage bucket</a> for storage
+APIs.
```

Comment:
This is confusing because "session storage APIs" sounds like a subset of "storage APIs". I think it'd be good to have two distinct terms. (Maybe changing the `"storage"` and `"session-storage"` types as well.)

Reply:
Let me know if you think of something good.

Reply:
I guess "persistent" has a different meaning. "local", maybe.

Reply:
I did think of that, but calling IDB a local storage API doesn't seem very clear either. Hmm.

Reply:
eduring, durable, lasting, long-lived?